test/cookie.test.js

```diff
@@ -847,3 +847,49 @@ test('should not decorate fastify, request and reply if no secret was provided',
     message: 'req.unsignCookie is not a function'
   })
 })
+
+test('dont add default @fastify/cookie hook-function to fastify if hook-option to false  ', (t) => {
+  t.plan(6)
+  const fastify = Fastify()
+  fastify.register(plugin, { hook: false })
+
+  for (const hook of ['preValidation', 'preHandler']) {
+    fastify.addHook(hook, (req, reply, done) => {
+      t.equal(req.cookies, null)
+      done()
+    })
+  }
+
+  fastify.addHook('preParsing', (req, reply, payload, done) => {
```

Comment:
You should be able to add this hook to the hook array above by defining:

```js
async (req) => {
 t.equal(req.cookies, null)
}
```

Reply:
what do you mean?

Reply:
added a suggestion to clarify